File: API/Features/Weather/Dtos/WeatherForecastGeneralDto.cs
using System.Collections.Generic; 

namespace WeatherForecast.WebApi.Features.Weather.Dtos
{
    public class WeatherForecastGeneralDto
    {
        public string SelectedCity { get; set; }
        public List<LinkDto> Links{ get; set; }
        public List<CityDto> Cities { get; set; }
        public List<WeatherForecastDto> WeatherForecast { get; set; }
    }
}

File: clientApp/src/types/WeatherForecastGeneralDto.ts
export default interface WeatherForecastGeneralDto {
    selectedCity: string,
    links: Array<LinkDto>,
    cities: Array<CityDto>,
    weatherForecast: Array<WeatherForecastDto>,
    errored: boolean
}

export interface LinkDto {
    rel: string,
    href: string
}

export interface CityDto {
    name: string,
    zipCode: string
}

export interface WeatherForecastDto {
    date: string,
    temperature: string,
    humidity: string,
    windSpeed: string
}